File: 540. Single Element in a Sorted Array/540. Single Element in a Sorted Array.swift
class Solution {
    func singleNonDuplicate(_ nums: [Int]) -> Int {
        var i = 0
        var lengthofArray = nums.count - 1 // get the length of the array
        while i < lengthofArray {
            if nums[i] != nums[i + 1]{
                return nums[i]
            } else {
                i += 2
            }
        }
        return nums[lengthofArray]
    }
}

File: 540. Single Element in a Sorted Array/540. Single Element in a Sorted Array.go
func singleNonDuplicate(nums []int) int {
    var l = len(nums)
    for i := 0; i < l - 2; i = i + 2 {
        if nums[i] != nums[i + 1] {
            return nums[i]
        }
    }
    return nums[l - 1]
}
